### ISSUE TITLE
Getting Tried to run command without establishing connection exception(Selenium v3.4.4 ,gecko driver v0.16.1)

### ISSUE THREAD

**Issue by prabhurohith** [mozilla/geckodriver#737]
## Firefox Version
FF Version : 53.0.2

## Platform
Windows 7

## Steps to reproduce -
- Launch the firefox browser using selenium 

Code Used:
```Java
String pathDriver = System.getProperty("user.dir");
					System.out.println("This is the directory" + pathDriver);
					String pathDriverNew = pathDriver + "/drivers/geckodriver.exe";

					System.setProperty("webdriver.gecko.driver", pathDriverNew);
					

					DesiredCapabilities capabilities = DesiredCapabilities.firefox();
					FirefoxOptions options = new FirefoxOptions();
								
					options.addPreference("log", "{level: trace}");
							
					capabilities.setCapability("marionette", true);
					capabilities.setCapability("moz:firefoxOptions", options);
							
								
					driver = new FirefoxDriver(capabilities);
					
```

## Trace Logs

```
1494840966868	geckodriver	INFO	Listening on 127.0.0.1:27193
1494840967840	geckodriver::marionette	INFO	Starting browser \\?\C:\Program Files (x86)\Mozilla Firefox\firefox.exe with args ["-marionette"]
1494840969576	addons.manager	ERROR	startup JfaaivlaeSd: [Exceptiocrni.p.t.  e"rCroomrp:o nreensto urrecteu:r/n/egdr ef/amiolduurles/eA dcdoodneM:a n0axg8e0r0.7j0s0m5,7  l(iNnSe_ E1R6R5O7R:_ INLSL_EERROR_NOT_INITIALIZED: AddonManager is not initialized
GAL_VALUE) [nsIFile.create]"  nsresult: "0x80070057 (NS_ERROR_ILLEGAL_VALUE)"  location: "JS frame :: resource://gre/modules/FileUtils.jsm :: FileUtils_getDir :: line 70"  data: no] Stack trace: FileUtils_getDir()@resource://gre/modules/FileUtils.jsm:70 < FileUtils_getFile()@resource://gre/modules/FileUtils.jsm:42 < validateBlocklist()@resource://gre/modules/AddonManager.jsm:671 < startup()@resource://gre/modules/AddonManager.jsm:834 < startup()@resource://gre/modules/AddonManager.jsm:3129 < observe()@resource://gre/components/addonManager.js:65
1494840974273	Marionette	INFO	Listening on port 62106
JavaScript error: resource://gre/modules/AddonManager.jsm, line 2570: NS_ERROR_NOT_INITIALIZED: AddonManager is not initialized
1494840975363	Marionette	WARN	TLS certificate errors will be ignored for this session
May 15, 2017 3:06:15 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
JavaScript warning: https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.24/browser.js, line 64284: unreachable code after return statement
JavaScript warning: https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.24/browser.js, line 64284: unreachable code after return statement
JavaScript error: resource://gre/modules/FileUtils.jsm, line 70: NS_ERROR_ILLEGAL_VALUE: Component returned failure code: 0x80070057 (NS_ERROR_ILLEGAL_VALUE) [nsIFile.create]
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type org.sikuli.util.SysJNA$WinKernel32
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type org.bridj.TimeT
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type org.bridj.TimeT$timeval_customizer
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type org.bridj.StructIO$DefaultCustomizer
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type org.bridj.TimeT$timeval
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type org.bridj.StructObject
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type org.bridj.NativeObject
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type org.bridj.AbstractIntegral
May 15, 2017 3:06:18 PM org.bridj.BridJ log
INFO: Registering type java.lang.Number 
```

**Comment by AutomatedTester** [comment]
I can see that a session is being created by your example so it is working as intended. if you have more information then please add it to the bug and we can see about reopening.

**Comment by prabhurohith** [comment]
The session is being created but the firefox browser closes as soon as it is launched ,the issue isn't being observed when i move to v0.16.0

**Comment by AutomatedTester** [comment]
seeing as 0.16.1 is the latest I suggest you move to that.

**Comment by andreastt** [comment]
The log also doesn’t appear to include trace-level entries because you add `{"log": {"level": "trace"}}` as a _preference_. It is a capability.

I think in Java you can set this using the `FirefoxOptions.setLogLevel(LogLevel.ALL)` class.

**Comment by prabhurohith** [comment]
@AutomatedTester (David) ,thanks i downgraded and upgraded it to 0.16.1 ,not getting the issue now ,but i get some issue on context click will open a ticket for same

**Comment by prabhurohith** [comment]
@andreastt :thanks the above command worked but is it possible to only get trace level logs using the above command as i can't see Trace method for the same

**Comment by Temp12d** [comment]
Hi,

Getting below webdriveException on firefox browser


Failureorg.openqa.selenium.WebDriverException: Tried to run command without establishing a connection Build info: version: 'unknown', revision: '1969d75', time: '2016-10-18 09:43:45 -0700' System info: host: 'egain1210', ip: '10.10.54.104', os.name: 'Windows 10', os.arch: 'x86', os.version: '10.0', java.version: '1.8.0_151' Driver info: org.openqa.selenium.firefox.FirefoxDriver Capabilities [{rotatable=false, raisesAccessibilityExceptions=false, marionette=true, firefoxOptions={args=[], prefs={}}, appBuildId=20160623154057, version=, platform=XP, proxy={}, command_id=1, specificationLevel=0, acceptSslCerts=false, browserVersion=47.0.1, platformVersion=10.0, XULappId={ec8030f7-c20a-464f-9b0e-13a3a9e97384}, browserName=Firefox, takesScreenshot=true, takesElementScreenshot=true, platformName=Windows_NT, device=desktop}] Session ID: e89fddda-010f-4f5b-a1f7-ecbbae9f5c22 *** Element info: {Using=xpath, value=//span[text()='print']} 	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:127) 	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:93) 	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:42) 	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:163) 	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:82) 	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:601) 	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:368) 	at org.openqa.selenium.remote.RemoteWebDriver.findElementByXPath(RemoteWebDriver.java:473) 	at org.openqa.selenium.By$ByXPath.findElement(By.java:361) 	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:360) 	at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:69) 	at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38) 	at com.sun.proxy.$Proxy12.toString(Unknown Source) 	at com.egain.framework.BasePageObjects.click(BasePageObjects.java:469) 	at com.egain.pageobjects.base.ee.AlbertArticlePage.performArticlePrintOperation(AlbertArticlePage.java:39) 	at com.egain.tests.ee.temp.testEE08_ArticleHeaderToolOperations(temp.java:115) ... Removed 30 stack frames
--


Can you please help me on this

**Comment by andreastt** [comment]
@Temp12d That’s an error on your end and not a bug in geckodriver. Please also don’t hijack existing issues.

**Comment by whimboo** [comment]
Please note the following line from the original log:

> 1494840967840	geckodriver::marionette	INFO	Starting browser \\?\C:\Program Files (x86)\Mozilla Firefox\firefox.exe with args ["-marionette"]

This is the problem we fixed for the 0.17 release. It's a dupe of issue #702

**Comment by andreastt** [comment]
Oh, well spotted.
